calico_kubernetes/logutils.py

```diff
@@ -46,3 +46,10 @@ def configure_logger(logger, log_level, root_logger=False, log_dir=LOG_DIR):
     logger.addHandler(file_hdlr)
     logger.addHandler(stdout_hdlr)
     logger.setLevel(log_level)
+
+class DockerIDLogger(logging.LoggerAdapter):
+    """
+    Process Logger messages to append the Docker ID.
+    """
+    def process(self, msg, kwargs):
+        return '[%s] %s' % (self.extra['docker_id'], msg), kwargs
```

Comment:
If we can avoid prepending the docker_id like this, I'd prefer it; the 'standard' log format is "TIMESTAMP OTHERSTUFF" and it's a bit of a pain to read and parse if it's otherwise.

I don't think you need this subclass; just use a LoggerAdapter, called with the arguments that you're currently using. The default implementation of process() is to add the contents of the `extra` dict to the `kwargs` from the log call.